aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1256,14 +1256,64 @@ static inline Tensor& bmm_out_or_baddbmm_(Tensor& self_or_result, const Tensor&
             && self_or_result.is_contiguous()) {
     at::native::_baddbmm_mkl_(self_or_result, batch1, batch2, beta, alpha);
   } else { // split along batch dimension
+    bool enable_multithreaded_bmm{false};
```

Comment:
Could you move this into the #else of the macro and make it const?
That would allow the compiler to completely prune the code in the conditions below: https://godbolt.org/z/8894Mdxd1

Reply:
Oh interesting. I did not realize that without `const` compiler would not 'const propagate'. Seems strange. But sure I can make the change.

Reply:
If you do "-Os" then it simplifies a lot. Nonetheless I will make the change.

Reply:
It does with optimization enabled. But not for debug versions from what I see.